core/shared/src/main/scala/scalaz/zio/IO.scala

```diff
@@ -0,0 +1,571 @@
+   * }}}
+   */
+  final def access: ZIO.AccessPartiallyApplied[IoR] =
+    ZIO.access
+
+  /** Effectfully accesses the environment of the effect.
+   */
+  final def accessM: ZIO.AccessMPartiallyApplied[IoR] =
+    ZIO.accessM
+
+  /** Given an environment `R`, returns a function that can supply the
+   * environment to programs that require it, removing their need for any
+   * specific environment.
+   *
+   * This is similar to dependency injection, and the `provide` function can be
+   * thought of as `inject`.
+   */
+  final def provide[E, A](r: IoR): ZIO[IoR, E, A] => IO[E, A] =
+    ZIO.provide(r)
+
+  /** Returns an effect that accesses the runtime, which can be used to
+   * (unsafely) execute tasks. This is useful for integration with
+   * non-functional code that must call back into functional code.
+   */
+  final def runtime: ZIO[IoR, Nothing, Runtime[IoR]] = ZIO.runtime
+
+  /** Returns an effect that is interrupted.
+   */
+  final val interrupt: UIO[Nothing] = ZIO.interrupt
+
+  /** Returns a effect that will never produce anything. The moral
+   * equivalent of `while(true) {}`, only without the wasted CPU cycles.
+   */
+  final val never: UIO[Nothing] = ZIO.never
+
+  /** Returns an effect that dies with the specified `Throwable`.
+   * This method can be used for terminating a fiber because a defect has been
+   * detected in the code.
+   */
+  final def die(t: Throwable): UIO[Nothing] = ZIO.die(t)
+
+  /** Returns an effect that dies with a [[java.lang.RuntimeException]] having the
+   * specified text message. This method can be used for terminating a fiber
+   * because a defect has been detected in the code.
+   */
+  final def dieMessage(message: String): UIO[Nothing] = ZIO.dieMessage(message)
+
+  /** Imports a total synchronous effect into a pure `ZIO` value.
+   * The effect must not throw any exceptions. If you wonder if the effect
+   * throws exceptions, then do not use this method, use [[Task.effect]],
+   * [[IO.effect]], or [[ZIO.effect]].
+   *
+   * {{{
+   * val nanoTime: UIO[Long] = IO.effectTotal(System.nanoTime())
+   * }}}
+   */
+  final def effectTotal[A](effect: => A): UIO[A] = ZIO.effectTotal(effect)
+
+  /** Imports a total synchronous effect into a pure `ZIO` value. This variant
+   * of `effectTotal` lets the impure code use the platform capabilities.
+   *
+   * The effect must not throw any exceptions. If you wonder if the effect
+   * throws exceptions, then do not use this method, use [[Task.effect]],
+   * [[IO.effect]], or [[ZIO.effect]].
+   *
+   * {{{
+   * val nanoTime: UIO[Long] = IO.effectTotal(System.nanoTime())
+   * }}}
+   */
+  final def effectTotalWith[A](effect: Platform => A): UIO[A] = ZIO.effectTotalWith(effect)
+
+  /** Returns an effect that yields to the runtime system, starting on a fresh
+   * stack. Manual use of this method can improve fairness, at the cost of
+   * overhead.
+   */
+  final val yieldNow: UIO[Unit] = ZIO.Yield
```

Comment:
For all these files, try just to delegate to ZIO methods, not any of the data constructors directly.